pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,226 @@
+	detailsFlag  bool
+}
+
+// NewListOptions creates a new ListOptions instance
+func NewListOptions() *ListOptions {
+	return &ListOptions{}
+}
+
+func (o *ListOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+// Complete completes ListOptions after they've been created
+func (o *ListOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+
+	o.DevfileList, err = o.clientset.RegistryClient.ListDevfileStacks("")
+	if err != nil {
+		return err
+	}
+
+	if o.DevfileList.DevfileRegistries == nil {
+		log.Warning("Please run 'odo preference registry add <registry name> <registry URL>' to add registry for listing devfile components\n")
+	}
+
+	return nil
+}
+
+// Validate validates the ListOptions based on completed values
+func (o *ListOptions) Validate() error {
+	if len(o.DevfileList.Items) == 0 {
+		return fmt.Errorf("no deployable components found")
+	}
+	return nil
+}
+
+// Run contains the logic for the command associated with ListOptions
+func (o *ListOptions) Run(ctx context.Context) (err error) {
+	o.printDevfileList(o.DevfileList.Items)
+	return nil
+}
+
+func NewCmdRegistry(name, fullName string) *cobra.Command {
+	o := NewListOptions()
+
+	var listCmd = &cobra.Command{
+		Use:     name,
+		Short:   "List all components from the Devfile registry",
+		Long:    "List all components from the Devfile registry",
+		Example: fmt.Sprintf(Example, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			genericclioptions.GenericRun(o, cmd, args)
+		},
+	}
+
+	clientset.Add(listCmd, clientset.REGISTRY)
+
+	// Flags
+	listCmd.Flags().StringVar(&o.filterFlag, "filter", "", "Filter based on the name of the component")
+	listCmd.Flags().StringVar(&o.devfileFlag, "devfile", "", "Only the specific Devfile component")
+	listCmd.Flags().StringVar(&o.registryFlag, "devfile-registry", "", "Only show components from the specific Devfile registry")
+	listCmd.Flags().BoolVar(&o.detailsFlag, "details", false, "Show details of each component")
+
+	// Add a defined annotation in order to appear in the help menu
+	listCmd.Annotations["command"] = "main"
+	listCmd.SetUsageTemplate(odoutil.CmdUsageTemplate)
+
+	return listCmd
+}
+
+func (o *ListOptions) printDevfileList(DevfileList []registry.DevfileStack) {
+
+	// Create the table and use our own style
+	t := table.NewWriter()
+
+	t.SetStyle(table.Style{
+		Box: table.BoxStyle{
+			PaddingLeft:  " ",
+			PaddingRight: " ",
+		},
+		Color: table.ColorOptions{
+			Header: text.Colors{text.FgHiGreen, text.Underline},
+		},
+		Format: table.FormatOptions{
+			Footer: text.FormatUpper,
+			Header: text.FormatUpper,
+			Row:    text.FormatDefault,
+		},
+		Options: table.Options{
+			DrawBorder:      false,
+			SeparateColumns: false,
+			SeparateFooter:  false,
+			SeparateHeader:  false,
+			SeparateRows:    false,
+		},
+	})
+	t.SetOutputMirror(log.GetStdout())
+
+	t.AppendHeader(table.Row{"NAME", "REGISTRY", "DESCRIPTION"})
+	t.SortBy([]table.SortBy{
```

Comment:
I think we want to order by registry in order of appareance in the config (the last in the list appearing first):

```
$ cat ~/.odo/preference.yaml
kind: Preference
apiversion: odo.dev/v1alpha1
OdoSettings:
  RegistryList:
  - Name: DefaultDevfileRegistry
    URL: https://registry.devfile.io
    secure: false
  - Name: MyRegistry
    URL: https://registry.stage.devfile.io
    secure: false
  Ephemeral: false
  ConsentTelemetry: true


$ odo registry
 NAME                          REGISTRY                DESCRIPTION                                 
 dotnet50                      DefaultDevfileRegistry  Stack with .NET 5.0                         
 dotnet50                      MyRegistry              Stack with .NET 5.0                         
 dotnet60                      DefaultDevfileRegistry  Stack with .NET 6.0                         
 dotnet60                      MyRegistry              Stack with .NET 6.0                         
 dotnetcore31                  DefaultDevfileRegistry  Stack with .NET Core 3.1                    
 dotnetcore31                  MyRegistry              Stack with .NET Core 3.1                    
```

should be

```
$ odo registry
 NAME                          REGISTRY                DESCRIPTION                                 
 dotnet50                      MyRegistry              Stack with .NET 5.0                         
 dotnet50                      DefaultDevfileRegistry  Stack with .NET 5.0                         
 dotnet60                      MyRegistry              Stack with .NET 6.0                         
 dotnet60                      DefaultDevfileRegistry  Stack with .NET 6.0                         
 dotnetcore31                  MyRegistry              Stack with .NET Core 3.1                    
 dotnetcore31                  DefaultDevfileRegistry  Stack with .NET Core 3.1                    
```